src/baldr/datetime.cc

```diff
@@ -13,28 +13,59 @@
 #include "midgard/logging.h"
 #include "midgard/util.h"
 
+namespace {
+// use a cache to store already constructed sys_info's since they aren't cheap
+template <typename TP>
+const date::sys_info&
+from_cache(const TP& tp,
+           const date::time_zone* tz,
+           std::unordered_map<const date::time_zone*, std::vector<date::sys_info>>& cache) {
+  // check if we have anything for this timezone in the cache
+  auto tz_it = cache.find(tz);
+  if (tz_it != cache.cend()) {
+    // we have something in the cache, see if one of the infos has the particular time point in range
+    auto st = date::floor<std::chrono::seconds>(tp.get_sys_time());
```

Comment:
Could `st` be named something more verbose?

Reply:
I was just following the patterns of the library it comes from. Tp=timepoint and st=system time. I think it's pretty clear from the function call get sys time no?